Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Mind Control and Turn Undead Blocks

# Changelog:
- Added new `Is NOT currently raising or creating an undead being` condition in `Turn Undead`, `Create Monstrous Undead`, `Turn yourself Undead`, `Turn yourself into an abomination` decisions. This condition should block using the said decisions at the same time.
- You can no longer use `Mind Control` on a few characters at the same time.

# Secret Changelog:
- Added `ask_for_dark_gift_used_flag` flag that prevents clearing of `raising_undead_flag` from the Lich King during "Ask for Dark Gift" decision of the Cult of the Damned.

# How to Test
**Since you're testing flags, make sure you know** [**Functions of `charinfo`**](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/Console-Tricks#charinfo
).
- Play any necromancer (like Scourge liches) and try to turn a few characters undead at the same time. Make sure you have no `raising_undead_flag` flag on your character after using the decision.
- Play any high level warlock (like Gul'dan) and try to `Mind Control` a few characters at the same time. Make sure you have no `control_mind_ongoing_flag` flag on your character after using the decision.
- Play Alexei Barov, set `raising_undead_flag` flag to Ner'zhul using console command `set_char_flag raising_undead_flag 10014`, use `Ask for Dark Gift` decisions, make sure Ner'zhul has `raising_undead_flag` flag and you have no `ask_for_dark_gift_used_flag` after using the decision.
